pilz_robot_programming/src/pilz_robot_programming/commands.py

```diff
@@ -201,6 +201,28 @@ def __str__(self):
         out_str += " reference: " + str(self._reference_frame)
         return out_str
 
+    def __eq__(self, other):
+        if isinstance(other, BaseCmd):
+            return self._goal == other._goal \
+                   and self._planner_id == other._planner_id \
+                   and self._planning_group == other._planning_group \
+                   and self._target_link == other._target_link \
+                   and self._vel_scale == other._vel_scale \
+                   and self._acc_scale == other._acc_scale \
+                   and self._relative == other._relative \
+                   and self._reference_frame == other._reference_frame
+
+        return NotImplemented
+
+    def __ne__(self, other):
+        x = self.__eq__(other)
+        if x is not NotImplemented:
+            return not x
+        return NotImplemented
+
+    def __hash__(self):
+        return hash(tuple(sorted(self.__dict__.items())))
```

Comment:
```suggestion
        return hash(tuple(map(lambda k: self.__dict__[k], sorted(self.__dict__.keys()))))
```
I did not test it, but this should make https://github.com/PilzDE/pilz_industrial_motion/pull/255/files#diff-288808388fd41c35fcc136672ed08a47R207-R213 not necessary

Reply:
I had to change the method. Did try to change and remove/replace code like you suggested anyway, but I didn't find a working solution. Either provide more precise input, of what to delete or replace, or test it yourself with the provided tests please.

Reply:
more or less implemented your change, you can check if you are OK with my implementation.